doc/examples/applications/plot_haar_extraction_selection_classification.py

```diff
@@ -119,46 +121,50 @@ def extract_feature_image(img, feature_type, feature_coord=None):
 
-###############################################################################
-# We can select the most important features by checking the cumulative sum of
-# the feature importance index; below, we keep features representing 70% of the
-# cumulative value which represent only 3% of the total number of features.
+###########################################################################
+# We can select the most important features by checking the cumulative sum
+# of the feature importance. In this example, we keep the features
+# representing 70% of the cumulative value (what corresponds to using only 3%
+# of the total number of features).
 
 cdf_feature_importances = np.cumsum(clf.feature_importances_[idx_sorted])
 cdf_feature_importances /= np.max(cdf_feature_importances)
 sig_feature_count = np.count_nonzero(cdf_feature_importances < 0.7)
 sig_feature_percent = round(sig_feature_count /
                             len(cdf_feature_importances) * 100, 1)
-print(('{} features, or {}%, account for 70% of branch points in the random '
-       'forest.').format(sig_feature_count, sig_feature_percent))
-
-# Select the most informative features
-selected_feature_coord = feature_coord[idx_sorted[:sig_feature_count]]
-selected_feature_type = feature_type[idx_sorted[:sig_feature_count]]
-# Note: we could select those features from the
-# original matrix X but we would like to emphasize the usage of `feature_coord`
-# and `feature_type` to recompute a subset of desired features.
-
-# Delay the computation and build the graph using dask
-X = delayed(extract_feature_image(img, selected_feature_type,
-                                  selected_feature_coord)
+print(('{} features, or {}%, account for 70% of branch points in the '
+       'random forest.').format(sig_feature_count, sig_feature_percent))
+
+# Select the determined number of most informative features
+feature_coord_sel = feature_coord[idx_sorted[:sig_feature_count]]
+feature_type_sel = feature_type[idx_sorted[:sig_feature_count]]
+# Note: it is also possible to select the features directly from the matrix X,
+# but we would like to emphasize the usage of `feature_coord` and `feature_type`
+# to recompute a subset of desired features.
+
+# Build the computational graph using Dask
+X = delayed(extract_feature_image(img, feature_type_sel, feature_coord_sel)
             for img in images)
-# Compute the result using the *threads* backend:
-# When computing all features, the Python GIL is acquired to process each ROI,
-# and this is where most of the time is spent, so multiprocessing is faster.
-# For this small subset, most of the time is spent on the feature computation
-# rather than the ROI scanning, and using threaded is *much* faster, because
-# we avoid the overhead of launching a new process.
+# Compute the result
 t_start = time()
-X = np.array(X.compute(scheduler='threads'))
+if sys.platform.startswith('win'):
+    X = np.array(X.compute(scheduler='synchronous'))
```

Comment:
Could you link to the specific dask docs page that talks about this? Is it necessary for threaded as well as multiprocessing? I almost would prefer not running this example at all in Windows than having the code structured like this.

Reply:
Threads seem to work, thanks for the suggestion!